eclass/elisp-common.eclass

```diff
@@ -287,6 +298,34 @@ elisp-need-emacs() {
 	return 0
 }
 
+# @FUNCTION: elisp-emacs-nativecomp
+# @RETURN: exit status of Emacs
+# @DESCRIPTION:
+# Output whether currently active Emacs was built with nativecomp.
+
+elisp-emacs-nativecomp() {
+	local cmd
+	if ! ver_test "${_ELISP_EMACS_VERSION}" -ge "${_EMACS_MAYBE_NATIVE_COMP}"; then
+		return 1
+	fi
+	cmd='(or (string-match-p "--with-nativecomp" system-configuration-options) (error "nativecomp not enabled"))'
+	return ${EMACS} ${EMACSFLAGS} -l "${cmd}"
+}
+
+# @FUNCTION: elisp-check-emacs-nativecomp
+# @USAGE: [version]
+# @DESCRIPTION:
+# Test if the eselected Emacs version is at least the version of
+# GNU Emacs specified in the NEED_EMACS variable, or die otherwise.
+
+elisp-check-emacs-nativecomp() {
+	if [[ -z ${_ELISP_EMACS_NATIVE_COMP} ]]; then
+		elisp-emacs-nativecomp
+		_ELISP_EMACS_NATIVE_COMP=$?
+	fi
+	return ${_ELISP_EMACS_NATIVE_COMP}
+}
```

Comment:
I wonder if two functions are necessary here. Looks like a single `elisp-emacs-nativecomp` (but with the caching variable) would be sufficient?

OTOH, if you model it along the lines of `elisp-emacs-version` and `elisp-emacs-check-version` then the "check" function should die upon failure.

Reply:
Yes, I wondered about the naming here. Basically I wanted to check once whether emacs had support built, store that, and then reuse it instead of checking each time, but I don't think this should fail, because there aren't any packages that strictly require nativecomp in the same way that they might require a certain minimum version of emacs. Maybe just have `elisp-emacs-nativecomp` do the caching itself and get rid of `elisp-emacs-check-nativecomp` altogether?